frontend/discourse/tests/unit/lib/lightbox/quote-image-test.js

```diff
@@ -0,0 +1,309 @@
+import Controller from "@ember/controller";
+import Service from "@ember/service";
+import { setupTest } from "ember-qunit";
+import { module, test } from "qunit";
+import sinon from "sinon";
+import quoteImage, { canQuoteImage } from "discourse/lib/lightbox/quote-image";
+import Draft from "discourse/models/draft";
+
+class ComposerStub extends Service {
+  model = { viewOpen: false };
+  openCalls = [];
+
+  async open(args) {
+    this.openCalls.push(args);
+  }
+}
+
+class StoreStub extends Service {
+  constructor() {
+    super(...arguments);
+    this.posts = new Map();
+    this.topics = new Map();
+  }
+
+  peekRecord(type, id) {
+    const key = id?.toString();
+
+    if (type === "post") {
+      return this.posts.get(key);
+    }
+
+    if (type === "topic") {
+      return this.topics.get(key);
+    }
+
+    return null;
+  }
+
+  async find(type, id) {
+    return this.peekRecord(type, id);
+  }
+}
+
+class AppEventsStub extends Service {
+  events = [];
+
+  trigger(...args) {
+    this.events.push(args);
+  }
+}
+
+module("Unit | Lib | lightbox | quote image", function (hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function () {
+    this.owner.unregister("service:composer");
+    this.owner.unregister("service:store");
+    this.owner.unregister("service:app-events");
+
+    this.owner.register("service:composer", ComposerStub);
+    this.owner.register("service:store", StoreStub);
+    this.owner.register("service:app-events", AppEventsStub);
+    this.owner.register("controller:topic", class extends Controller {});
+
+    this.composer = this.owner.lookup("service:composer");
+    this.store = this.owner.lookup("service:store");
+    this.appEvents = this.owner.lookup("service:app-events");
+    this.topicController = this.owner.lookup("controller:topic");
+
+    this.topic = { id: 100, draft_key: "topic_100", draft_sequence: 5 };
+    this.post = {
+      id: 321,
+      post_number: 2,
+      username: "alice",
+      name: "Alice",
+      topic_id: this.topic.id,
+      topicId: this.topic.id,
+      topic: this.topic,
+    };
+
+    this.store.posts.set(this.post.id.toString(), this.post);
```

Comment:
I think this file generally just needs a lot of cleanup as I was saying yesterday, store is no longer needed since all of this stuff is sent via slideData now

Reply:
thanks will take care of this now!